Lesson15/Lesson15/gokce/script.js

```diff
@@ -0,0 +1,391 @@
+     (or a loop) to swap all occurrences of oldChar with newChar.
+   - Log the result.
+*/
+function replaceCharacter(str,oldChar,newChar){
+  console.log(str.replaceAll(oldChar,newChar));
+}
+replaceCharacter("arrangements","e",".");
+/*
+19. Title Case a Sentence
+   - Define a function `titleCase(sentence)` that:
+     - Splits the sentence by spaces
+     - Uppercases the first letter of each word
+     - Joins them back
+   - Log the transformed string.
+*/
+function titleCase(sentence){
+  let sentenceArray=sentence.split(" ");
+  sentenceArray=sentenceArray.map(word=>word.charAt(0).toUpperCase()+word.slice(1).toLowerCase());
+  return sentenceArray.join(" ");
+  
+}
+console.log(titleCase("Splits the sentence by spaces Uppercases the first letter of each word Joins them back"));
+/*
+20. Switch: Traffic Light
+   - Define a function `trafficLight(color)` that uses a switch statement:
+     - "red" -> log: "Stop" 
+     - "yellow" -> log: "Caution"
+     - "green" -> log: "Go"
+     - anything else -> "Invalid color"
+*/
+function trafficLight(color){
+  switch(color){
+    case "red":      console.log("Stop");
+      break;
+    case "yellow":
+      console.log("Caution");
+      break;
+    case "green":
+      console.log("Go");
+      break;
+    default:
+      console.log("Invalid color");
+  }
+}
+trafficLight("red");
+trafficLight("blue");
+/*
+21. Check String Length (if-else)
+   - Define a function `isLongString(str)` that checks if the string length
+     is more than 10.
+   - Log "Long string" or "Short string".
+*/
+function isLongString(str) {
+  if(str.length>10){
+    console.log("Long string");
+  }else{
+    console.log("Short string");
+  }
+}
+isLongString("string");
+/*
+22. Convert to Lowercase Then Check
+   - Define a function `isSpam(text)` that converts the text to lowercase
+     and checks if it includes "spam".
+   - If it does, log "This text is spam."
+   - Otherwise, log "This text is not spam."
+*/
+function isSpam(text){
+  let textCheck=text.toLowerCase();
+  if(textCheck.includes("spam")){
+    console.log("This text is spam.");
+}else{
+  console.log("This text is not spam.");
+}}
+isSpam("This text is spam.");
+/*
+23. Extract Initials
+   - Define a function `getInitials(fullName)` that uses .split() to get each name part,
+     then logs the capitalized first letter of each.
+   - Example: "John Doe" -> "J.D."
+*/
+function getInitials(fullName){
+let fname=fullName.split(" ");
+let inialsTogether=fname.map(name=>name.charAt(0).toUpperCase());
+return inialsTogether.join(".");
+}
+console.log(getInitials("aysel gökçe kurt"));
+/*
+24. Switch: Month to Season
+   - Define a function `getSeason(monthNum)` (1-12). Use switch or if-else:
+     - 12, 1, 2  -> "Winter"
+     - 3, 4, 5   -> "Spring"
+     - 6, 7, 8   -> "Summer"
+     - 9, 10, 11 -> "Autumn"
+   - Log the season or "Invalid month" if out of range.
+*/
+function getSeason(monthNum){
+  switch(monthNum){
+    case 12,1,2:"Winter";
```

Comment:
Don't forget to add a return statement before the result, e.g.
```suggestion
    case 12,1,2:
     return "Winter";
```

Since return is used, remove 'break' keywords.